Dockertools unit test failed on OSX

**Kubernetes version**:

master branch.

**Environment**:

OSX.

**What happened**:

```
go test -v ./pkg/kubelet/dockertools
...
=== RUN   TestDeleteContainerRemoveDeadContainer
--- FAIL: TestDeleteContainerRemoveDeadContainer (0.00s)
	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is anything, is anything, is anything)
	controller.go:165: aborting test due to missing call(s)
=== RUN   TestGarbageCollectNetworkTeardown
--- FAIL: TestGarbageCollectNetworkTeardown (0.00s)
	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is anything, is anything, is equal to {docker 1867})
	controller.go:165: aborting test due to missing call(s)
=== RUN   TestGarbageCollectZeroMaxContainers
--- FAIL: TestGarbageCollectZeroMaxContainers (0.00s)
	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is anything, is anything, is anything)
	controller.go:165: aborting test due to missing call(s)
=== RUN   TestGarbageCollectNoMaxPerPodContainerLimit
--- FAIL: TestGarbageCollectNoMaxPerPodContainerLimit (0.00s)
	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is anything, is anything, is anything)
	controller.go:165: aborting test due to missing call(s)
=== RUN   TestGarbageCollectNoMaxLimit
--- FAIL: TestGarbageCollectNoMaxLimit (0.00s)
	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is anything, is anything, is anything)
	controller.go:165: aborting test due to missing call(s)

=== RUN   TestSyncPodWithRestartPolicy
E0220 21:50:34.208620   99748 docker_manager.go:1756] Can't make a ref to pod foo, container succeeded: 'selfLink was empty, can't make reference'
E0220 21:50:34.208740   99748 docker_manager.go:1756] Can't make a ref to pod foo, container failed: 'selfLink was empty, can't make reference'
E0220 21:50:34.208904   99748 docker_manager.go:1756] Can't make a ref to pod foo, container failed: 'selfLink was empty, can't make reference'
E0220 21:50:34.209048   99748 docker_manager.go:1565] unable to access pod data from container: no pod information labels on Docker container
--- FAIL: TestSyncPodWithRestartPolicy (0.00s)
        Error Trace:    docker_test.go:50
			docker_manager_test.go:943
	Error:		Received unexpected error "expected []string{\"inspect_container\", \"inspect_container\", \"stop\"}, got []string{\"inspect_container\", \"stop\"}"

=== RUN   TestGetPodStatusNoSuchContainer
E0220 21:50:34.222198   99748 docker_manager.go:2702] Unable to inspect container "nosuchcontainer": Error: No such container: nosuchcontainer
E0220 21:50:34.222567   99748 docker_manager.go:1756] Can't make a ref to pod foo, container POD: 'selfLink was empty, can't make reference'
E0220 21:50:34.222790   99748 docker_manager.go:1889] ResolvConfPath is empty.
E0220 21:50:34.222962   99748 docker_manager.go:1756] Can't make a ref to pod foo, container nosuchcontainer: 'selfLink was empty, can't make reference'
--- FAIL: TestGetPodStatusNoSuchContainer (0.00s)
        Error Trace:    docker_test.go:50
			docker_manager_test.go:1837
	Error:		Received unexpected error "expected []string{\"inspect_container\", \"create\", \"start\", \"inspect_container\", \"inspect_container\", \"create\", \"start\", \"inspect_container\"}, got []string{\"create\", \"start\", \"inspect_container\", \"inspect_container\", \"create\", \"start\", \"inspect_container\"}"

=== RUN   TestSyncPodDeadInfraContainerTeardown
E0220 21:50:34.223867   99748 docker_manager.go:1756] Can't make a ref to pod foo, container POD: 'selfLink was empty, can't make reference'
E0220 21:50:34.223980   99748 docker_manager.go:1889] ResolvConfPath is empty.
E0220 21:50:34.224100   99748 docker_manager.go:1756] Can't make a ref to pod foo, container nosuchcontainer: 'selfLink was empty, can't make reference'
--- FAIL: TestSyncPodDeadInfraContainerTeardown (0.00s)
        Error Trace:    docker_test.go:50
			docker_manager_test.go:1884
	Error:		Received unexpected error "expected []string{\"inspect_container\", \"create\", \"start\", \"inspect_container\", \"inspect_container\", \"create\", \"start\", \"inspect_container\"}, got []string{\"create\", \"start\", \"inspect_container\", \"inspect_container\", \"create\", \"start\", \"inspect_container\"}"

	controller.go:158: missing call(s) to *testing.MockNetworkPlugin.TearDownPod(is equal to new, is equal to foo, is anything)
	controller.go:165: aborting test due to missing call(s)
```

cc/ @yujuhong @Random-Liu @freehan

--- a/pkg/kubelet/dockertools/docker_manager_unsupported.go
+++ b/pkg/kubelet/dockertools/docker_manager_unsupported.go
@@ -47,7 +47,7 @@ func containerProvidesPodIP(containerName string) bool {
 }
 
 func containerIsNetworked(containerName string) bool {
-	return false
+	return containerName == PodInfraContainerName
 }
 
 // Returns nil as both Seccomp and AppArmor security options are not valid on Windows

